### DISCUSSION TITLE
How to do a proper CSS override for a page

### DISCUSSION THREAD

**Discussion by thdoan** [crxjs/chrome-extension-tools#613] [Q&A] [answered]
I'm trying to add a CSS override for the Options page (1), but I cannot do it without relying on `!important` since the injected CSS is below it (2):

![image](https://user-images.githubusercontent.com/4110567/208585480-02760146-0f54-4f92-9f11-94cfdde73f93.png)

Is there a way to inject the CSS above the `<style>` tag? Or how would you implement a CSS override for a page?

I'm using Svelte + Vite.

**Comment by abhishekZom** [comment]
You can dynamically inject a link tag before the style tag.

What CSS are you trying to override btw ?

Your question is not very clear.

Can you elaborate a little more ?

**Comment by thdoan** [comment]
@abhishekZom I'm trying to override body margin set by Tailwind CSS, which has this inside `Options-bedbe114.css`:
```
body {
  margin: 0;
}
```
I want to override only on the Options page by adding this to `options.html`:
```
<style>
body {
  margin: 1rem;
}
</style>
```
However, since  `Options-bedbe114.css` is injected below `<style>`  the override doesn't work. How do I configure it so that  `Options-bedbe114.css` is injected above `<style>` in `options.html`?

**Reply by abhishekZom** [reply]
In that case you won't be able to. Using `!important` is fine here.

**Comment by thdoan** [comment, marked as answer]
@abhishekZom thank you for your help. I'll accept this as the answer...

> @thdoan In that case you won't be able to. Using `!important` is fine here.
